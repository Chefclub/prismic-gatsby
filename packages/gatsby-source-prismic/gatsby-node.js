module.exports = require('./dist/gatsby-source-prismic')
