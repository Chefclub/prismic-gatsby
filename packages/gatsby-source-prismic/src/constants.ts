export const WEBHOOK_SECRET_MISMATCH_MSG =
  'A webhook was received, but the webhook secret did not match the webhook secret provided in the plugin options. If this is unexpected, verify that the `webhookSecret` plugin option matches the webhook secret in your Prismic repository.'

export const WEBHOOK_TEST_TRIGGER_SUCCESS_MSG =
  'Success! Received a test trigger webhook.'
