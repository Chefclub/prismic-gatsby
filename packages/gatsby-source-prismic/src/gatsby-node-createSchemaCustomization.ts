import * as RTE from 'fp-ts/ReaderTaskEither'
import { pipe, flow, constVoid } from 'fp-ts/function'
import {
  Dependencies,
  createBaseTypes,
  registerCustomTypes,
  registerAllDocumentTypesType,
} from 'gatsby-prismic-core'

/**
 * To be executed in the `createSchemaCustomization` stage.
 */
export const createSchemaCustomization: RTE.ReaderTaskEither<
  Dependencies,
  never,
  void
> = pipe(
  RTE.ask<Dependencies>(),
  RTE.chain(createBaseTypes),
  RTE.chain(flow(registerCustomTypes, RTE.chain(registerAllDocumentTypesType))),
  RTE.map(constVoid),
)
