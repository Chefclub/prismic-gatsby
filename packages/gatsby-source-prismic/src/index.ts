export * from './types'

export * from './constants'

export * from './lib/deserializeTypePathsStore'
export * from './lib/reportInfo'
export * from './lib/reportWarning'
export * from './lib/serializePath'
export * from './lib/sprintf'
export * from './lib/throwError'
