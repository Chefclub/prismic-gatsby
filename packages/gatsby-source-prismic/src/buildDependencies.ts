import * as gatsby from 'gatsby'

import { GLOBAL_TYPE_PREFIX } from './constants'
import { Dependencies, PluginOptions } from './types'
import { createNodeHelpers } from './lib/nodeHelpers'

export const buildDependencies = (
  gatsbyContext: gatsby.SourceNodesArgs | gatsby.CreateSchemaCustomizationArgs,
  pluginOptions: PluginOptions,
): Dependencies => ({
  pluginOptions,
  webhookBody: gatsbyContext.webhookBody,
  createNode: gatsbyContext.actions.createNode,
  createTypes: gatsbyContext.actions.createTypes,
  touchNode: gatsbyContext.actions.touchNode,
  deleteNode: gatsbyContext.actions.deleteNode,
  reportInfo: gatsbyContext.reporter.info,
  reportWarning: gatsbyContext.reporter.warn,
  buildUnionType: gatsbyContext.schema.buildUnionType,
  buildObjectType: gatsbyContext.schema.buildObjectType,
  buildEnumType: gatsbyContext.schema.buildEnumType,
  getNode: gatsbyContext.getNode,
  getNodes: gatsbyContext.getNodes,
  cache: gatsbyContext.cache,
  globalNodeHelpers: createNodeHelpers({
    typePrefix: GLOBAL_TYPE_PREFIX,
    createNodeId: gatsbyContext.createNodeId,
    createContentDigest: gatsbyContext.createContentDigest,
  }),
  nodeHelpers: createNodeHelpers({
    typePrefix: [GLOBAL_TYPE_PREFIX, pluginOptions.typePrefix]
      .filter(Boolean)
      .join(' '),
    fieldPrefix: GLOBAL_TYPE_PREFIX,
    createNodeId: gatsbyContext.createNodeId,
    createContentDigest: gatsbyContext.createContentDigest,
  }),
})
