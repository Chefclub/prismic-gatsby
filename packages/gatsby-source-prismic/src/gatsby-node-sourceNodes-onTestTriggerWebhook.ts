import * as RTE from 'fp-ts/ReaderTaskEither'

import { WEBHOOK_TEST_TRIGGER_SUCCESS_MSG } from './constants'
import { reportInfo } from './lib/reportInfo'
import { Dependencies } from './types'

export const onTestTriggerWebhook: RTE.ReaderTaskEither<
  Dependencies,
  never,
  void
> = reportInfo(WEBHOOK_TEST_TRIGGER_SUCCESS_MSG)
