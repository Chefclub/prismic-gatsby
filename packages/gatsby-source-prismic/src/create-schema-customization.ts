import * as gatsby from 'gatsby'
import * as RTE from 'fp-ts/ReaderTaskEither'
import * as E from 'fp-ts/Either'
import { pipe, constVoid } from 'fp-ts/function'

import { createBaseTypes } from './lib/createBaseTypes'
import { registerCustomTypes } from './lib/registerCustomTypes'
import { registerAllDocumentTypesType } from './lib/registerAllDocumentTypesType'
import { throwError } from './lib/throwError'

import { Dependencies, PluginOptions } from './types'
import { buildDependencies } from './buildDependencies'

export const createSchemaCustomization: NonNullable<
  gatsby.GatsbyNode['createSchemaCustomization']
> = async (
  gatsbyContext: gatsby.CreateSchemaCustomizationArgs,
  pluginOptions: PluginOptions,
) =>
  pipe(
    await RTE.run(
      createSchemaCustomizationProgram,
      buildDependencies(gatsbyContext, pluginOptions),
    ),
    E.fold(throwError, constVoid),
  )

/**
 * To be executed in the `createSchemaCustomization` stage.
 */
const createSchemaCustomizationProgram: RTE.ReaderTaskEither<
  Dependencies,
  never,
  void
> = pipe(
  RTE.ask<Dependencies>(),
  RTE.chain(createBaseTypes),
  RTE.bind('types', registerCustomTypes),
  RTE.chain((scope) => registerAllDocumentTypesType(scope.types)),
  RTE.map(constVoid),
)
