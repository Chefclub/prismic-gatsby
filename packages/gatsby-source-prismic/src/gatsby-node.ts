import * as gatsby from 'gatsby'
import * as RTE from 'fp-ts/ReaderTaskEither'
import * as E from 'fp-ts/Either'
import { pipe } from 'fp-ts/function'
import * as D from 'io-ts/Decoder'
import { PluginOptionsD } from 'gatsby-prismic-core'

import { buildDependencies } from './buildDependencies'
import { sourceNodes as sourceNodesProgram } from './gatsby-node-sourceNodes'
import { createSchemaCustomization as createSchemaCustomizationProgram } from './gatsby-node-createSchemaCustomization'

export const sourceNodes: NonNullable<gatsby.GatsbyNode['sourceNodes']> = (
  gatsbyContext: gatsby.SourceNodesArgs,
  pluginOptions: gatsby.PluginOptions,
) =>
  pipe(
    PluginOptionsD.decode(pluginOptions),
    E.fold(
      (error) => gatsbyContext.reporter.panic(new Error(D.draw(error))),
      (pluginOptions) =>
        RTE.run(
          sourceNodesProgram,
          buildDependencies(gatsbyContext, pluginOptions),
        ),
    ),
  )

export const createSchemaCustomization: NonNullable<
  gatsby.GatsbyNode['createSchemaCustomization']
> = (
  gatsbyContext: gatsby.CreateSchemaCustomizationArgs,
  pluginOptions: gatsby.PluginOptions,
) =>
  pipe(
    PluginOptionsD.decode(pluginOptions),
    E.fold(
      (error) => gatsbyContext.reporter.panic(new Error(D.draw(error))),
      (pluginOptions) =>
        RTE.run(
          createSchemaCustomizationProgram,
          buildDependencies(gatsbyContext, pluginOptions),
        ),
    ),
  )
