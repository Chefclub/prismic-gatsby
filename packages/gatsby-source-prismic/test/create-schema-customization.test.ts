import { createSchemaCustomization } from '../src/gatsby-node'
import { gatsbyContext, pluginOptions } from './__fixtures__/gatsby'

beforeEach(() => {
  jest.clearAllMocks()
})

describe('createSchemaCustomization', () => {
  test('creates types', async () => {
    // @ts-expect-error - Partial gatsbyContext provided
    await createSchemaCustomization(gatsbyContext, pluginOptions)

    expect(gatsbyContext.actions.createTypes).toMatchSnapshot()
  })
})
