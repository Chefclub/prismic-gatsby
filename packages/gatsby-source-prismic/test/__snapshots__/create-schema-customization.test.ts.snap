// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createSchemaCustomization creates types 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Array [
        Object {
          "config": Object {
            "name": "PrismicLinkTypes",
            "values": Object {
              "Any": Object {},
              "Document": Object {},
              "Media": Object {},
              "Web": Object {},
            },
          },
          "kind": "ENUM",
        },
        Object {
          "config": Object {
            "fields": Object {
              "height": "Int!",
              "width": "Int!",
            },
            "name": "PrismicImageDimensionsType",
          },
          "kind": "OBJECT",
        },
      ],
    ],
    Array [
      Object {
        "config": Object {
          "fields": Object {
            "document": Object {
              "extensions": Object {
                "link": Object {},
              },
              "resolve": [Function],
              "type": "PrismicPrefixAllDocumentTypes",
            },
            "id": "ID",
            "isBroken": "Boolean",
            "lang": "String",
            "link_type": "PrismicLinkTypes",
            "raw": Object {
              "resolve": [Function],
              "type": "JSON",
            },
            "size": "Int",
            "slug": "String",
            "tags": "[String]",
            "target": "String",
            "type": "String",
            "uid": "String",
            "url": Object {
              "resolve": [Function],
              "type": "String",
            },
          },
          "name": "PrismicPrefixLinkType",
        },
        "kind": "OBJECT",
      },
    ],
    Array [
      Object {
        "config": Object {
          "fields": Object {
            "html": Object {
              "resolve": [Function],
              "type": "String",
            },
            "raw": Object {
              "resolve": [Function],
              "type": "JSON",
            },
            "text": Object {
              "resolve": [Function],
              "type": "String",
            },
          },
          "name": "PrismicStructuredTextType",
        },
        "kind": "OBJECT",
      },
    ],
    Array [
      Object {
        "config": Object {
          "fields": Object {
            "html": Object {
              "resolve": [Function],
              "type": "String",
            },
            "raw": Object {
              "resolve": [Function],
              "type": "JSON",
            },
            "text": Object {
              "resolve": [Function],
              "type": "String",
            },
          },
          "name": "PrismicStructuredTextType",
        },
        "kind": "OBJECT",
      },
    ],
    Array [
      Object {
        "config": Object {
          "fields": Object {
            "html": Object {
              "resolve": [Function],
              "type": "String",
            },
            "raw": Object {
              "resolve": [Function],
              "type": "JSON",
            },
            "text": Object {
              "resolve": [Function],
              "type": "String",
            },
          },
          "name": "PrismicStructuredTextType",
        },
        "kind": "OBJECT",
      },
    ],
    Array [
      Array [
        Object {
          "config": Object {
            "fields": Object {
              "published_on": Object {
                "extensions": Object {
                  "dateformat": Object {},
                },
                "type": "Date",
              },
            },
            "name": "PrismicPrefixPageMainNewsPostPrimaryType",
          },
          "kind": "OBJECT",
        },
      ],
    ],
    Array [
      Object {
        "config": Object {
          "fields": Object {
            "alt": "String",
            "copyright": "String",
            "dimensions": "PrismicImageDimensionsType",
            "fixed": Object {
              "args": Object {
                "height": Object {
                  "type": "Int",
                },
                "imgixParams": Object {
                  "defaultValue": Object {},
                  "type": "ImgixUrlParamsInput",
                },
                "placeholderImgixParams": Object {
                  "defaultValue": Object {},
                  "type": "ImgixUrlParamsInput",
                },
                "width": Object {
                  "defaultValue": 400,
                  "type": "Int",
                },
              },
              "resolve": [Function],
              "type": "PrismicPrefixImageFixedType",
            },
            "fluid": Object {
              "args": Object {
                "imgixParams": Object {
                  "defaultValue": Object {},
                  "type": "ImgixUrlParamsInput",
                },
                "maxHeight": Object {
                  "type": "Int",
                },
                "maxWidth": Object {
                  "defaultValue": 800,
                  "type": "Int",
                },
                "placeholderImgixParams": Object {
                  "defaultValue": Object {},
                  "type": "ImgixUrlParamsInput",
                },
                "srcSetBreakpoints": Object {
                  "type": "[Int]",
                },
              },
              "resolve": [Function],
              "type": "PrismicPrefixImageFluidType",
            },
            "localFile": Object {
              "extensions": Object {
                "link": Object {},
              },
              "type": "File",
            },
            "url": Object {
              "args": Object {
                "imgixParams": Object {
                  "defaultValue": Object {},
                  "type": "ImgixUrlParamsInput",
                },
              },
              "resolve": [Function],
              "type": "String",
            },
          },
          "name": "PrismicPrefixImageType",
        },
        "kind": "OBJECT",
      },
    ],
    Array [
      Array [
        Object {
          "config": Object {
            "fields": Object {
              "text": "PrismicStructuredTextType",
            },
            "name": "PrismicPrefixPageBodyTextPrimaryType",
          },
          "kind": "OBJECT",
        },
      ],
    ],
    Array [
      Array [
        Object {
          "config": Object {
            "extensions": Object {
              "infer": false,
            },
            "fields": Object {
              "primary": "PrismicPrefixPageMainNewsPostPrimaryType",
              "slice_label": "String",
              "slice_type": "String!",
            },
            "name": "PrismicPrefixPageMainNewsPost",
          },
          "kind": "OBJECT",
        },
      ],
    ],
    Array [
      Array [
        Object {
          "config": Object {
            "fields": Object {
              "caption": "PrismicStructuredTextType",
              "image": "PrismicPrefixImageType",
              "orientation": "Boolean",
            },
            "name": "PrismicPrefixPageBodyImagesItemType",
          },
          "kind": "OBJECT",
        },
      ],
    ],
    Array [
      Object {
        "config": Object {
          "name": "PrismicPrefixPageMainSlicesType",
          "resolveType": [Function],
          "types": Array [
            "PrismicPrefixPageMainNewsPost",
          ],
        },
        "kind": "UNION",
      },
    ],
    Array [
      Array [
        Object {
          "config": Object {
            "extensions": Object {
              "infer": false,
            },
            "fields": Object {
              "items": "[PrismicPrefixPageBodyImagesItemType]",
              "slice_label": "String",
              "slice_type": "String!",
            },
            "name": "PrismicPrefixPageBodyImages",
          },
          "kind": "OBJECT",
        },
        Object {
          "config": Object {
            "extensions": Object {
              "infer": false,
            },
            "fields": Object {
              "primary": "PrismicPrefixPageBodyTextPrimaryType",
              "slice_label": "String",
              "slice_type": "String!",
            },
            "name": "PrismicPrefixPageBodyText",
          },
          "kind": "OBJECT",
        },
      ],
    ],
    Array [
      Object {
        "config": Object {
          "name": "PrismicPrefixPageBodySlicesType",
          "resolveType": [Function],
          "types": Array [
            "PrismicPrefixPageBodyImages",
            "PrismicPrefixPageBodyText",
          ],
        },
        "kind": "UNION",
      },
    ],
    Array [
      Object {
        "config": Object {
          "fields": Object {
            "body": "[PrismicPrefixPageBodySlicesType]",
            "main": "[PrismicPrefixPageMainSlicesType]",
            "meta_description": "String",
            "meta_title": "String",
            "parent": "PrismicPrefixLinkType",
            "title": "PrismicStructuredTextType",
          },
          "name": "PrismicPrefixPageDataType",
        },
        "kind": "OBJECT",
      },
    ],
    Array [
      Object {
        "config": Object {
          "extensions": Object {
            "infer": false,
          },
          "fields": Object {
            "_previewable": Object {
              "resolve": [Function],
              "type": "ID!",
            },
            "data": "PrismicPrefixPageDataType",
            "dataRaw": Object {
              "resolve": [Function],
              "type": "JSON!",
            },
            "first_publication_date": Object {
              "extensions": Object {
                "dateformat": Object {},
              },
              "type": "Date!",
            },
            "href": "String!",
            "lang": "String!",
            "last_publication_date": Object {
              "extensions": Object {
                "dateformat": Object {},
              },
              "type": "Date!",
            },
            "prismicId": "ID!",
            "tags": "[String!]!",
            "type": "String!",
            "uid": "String",
            "url": Object {
              "resolve": [Function],
              "type": "String",
            },
          },
          "interfaces": Array [
            "Node",
          ],
          "name": "PrismicPrefixPage",
        },
        "kind": "OBJECT",
      },
    ],
    Array [
      Object {
        "config": Object {
          "name": "PrismicPrefixAllDocumentTypes",
          "types": Array [
            "PrismicPrefixPage",
          ],
        },
        "kind": "UNION",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
